api/target/kusttarget.go

```diff
@@ -94,10 +94,7 @@ func loadKustFile(ldr ifc.Loader) ([]byte, error) {
 	}
 	switch match {
 	case 0:
-		return nil, fmt.Errorf(
-			"unable to find one of %v in directory '%s'",
-			commaOr(quoted(pgmconfig.RecognizedKustomizationFileNames())),
-			ldr.Root())
+		return nil, NewErrMissingKustomization(ldr.Root())
```

Comment:
+1 on this

Shall we also create an error type for the `default` case?

Reply:
Not right now, because all we need to handle differently is the case where the K file is missing.

The case of multiple K files will be handled as before (i.e. error should bubble up and fail the build).

Reply:
p.s. keeping the error type private per the advice of
https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully
which makes sense to me.  just expose an error testing method.